### ISSUE TITLE
Update "build directory" terminology in help and docs

### ISSUE THREAD

**Issue by tsibley** [nextstrain/cli#371]
This terminology was left as-is in https://github.com/nextstrain/cli/pull/355 (`nextstrain-pathogen.yaml`) even as it got a little more nuanced: it's no longer "the directory which was passed to `nextstrain build`." The term "build" has also drifted out of phase from terminology in the wider ecosystem.

Review and revise existing usages to clarify them and extend them to include the behaviour around `nextstrain-pathogen.yaml`.

There's a pretty good description of the behaviour in the [changelog for 8.2.0](https://github.com/nextstrain/cli/blob/14a392eaf40f779a7ef9431bbe0a116e93d98580/CHANGES.md#820-6-february-2024):

https://github.com/nextstrain/cli/blob/14a392eaf40f779a7ef9431bbe0a116e93d98580/CHANGES.md?plain=1#L52-L107

Motivated by @jameshadfield's [commentary](https://github.com/nextstrain/cli/pull/370#discussion_r1615357535) in review of a separate change.

**Comment by jameshadfield** [comment]
Here's some notes I made myself which may help others while the docs are forthcoming.

Example: we're in `avian-flu` running `nextstrain --docker ingest`. 

|                        | Without nextstrain-pathogen.yaml | With nextstrain-pathogen.yaml |
|------------------------|----------------------------------|-------------------------------|
| `/nextstrain/build` is | ingest                           | avian-flu                     |
| Starting directory     | `/nextstrain/build`              | `/nextstrain/build/ingest`    |

If we're in `avian-flu/ingest` and we run `nextstrain --docker .` the results are identical to above.

If we're in `avian-flu` and we run `nextstrain --docker .` then the presence of `nextstrain-pathogen.yaml` makes no difference.

**When does this matter?**

If you are using relative paths which track back to a higher directory than the one your workflow is in. For instance, `avian-flu/ingest/Snakemake` refers to the fauna location as `../fauna` ([src](https://github.com/nextstrain/avian-flu/blob/5f1ffda27148fcde1189c5576a44afbf6f21187d/ingest/Snakefile#L1)) and the repo doesn't define `nextstrain-pathogen.yaml` . Running this outside of docker doesn't work, as fauna isn't typically located at `avian-flu/fauna` . Using `nextstrain --docker ingest` works as we have `nextstrain/fauna` and `nextstrain/build` is the `ingest` dir.

**Comment by jameshadfield** [comment]
Another push for this - today I ran into it using [nextstrain/dengue](https://github.com/nextstrain/dengue/), which has a `nextstrain-pathogen.yaml`
```
cd phylogenetic
nextstrain build --aws-batch ... --download 'auspice/*' .
```

This doesn't work because the download paths need to be relative to the top-level of the pathogen repo, i.e. `--download 'phylogenetic/auspice/*'`. The [current docs](https://docs.nextstrain.org/projects/cli/en/stable/commands/build/#cmdoption-nextstrain-build-cancel) are out-of-date, saying:

> `<directory>` Path to pathogen build directory.

> `--download <pattern>` ... Patterns should be relative to the build directory.

I'm happy to contribute to the docs but I'm almost certain @tsibley will want to author these parts

**Comment by tsibley** [comment]
I wonder if the best fix for this particular case is making `--download` (and related options) aware of the working directory vs. pathogen directory. For example, `--download 'auspice/*'` would internally transform `auspice/*` into `phylogenetic/auspice/*`. I think this keeps the option simpler to explain and more intuitive since paths remain relative to the directory you provide to `nextstrain build`.

**Comment by jameshadfield** [comment]
Yeah, relative to the provided `<directory>` was my expectation, so I'd support this change. (And I think this both _was_ the situation before the recent changes related to separate analysis directories, and still _is_ the case if we don't have a `nextstrain-pathogen.yaml`.)

However we still need to convey what's going on here via help/docs messages, and the different directories that will be uploaded / bound. For instance, that command will log things like 
```
[batch] [...]   adding: phylogenetic/results/
```

which is confusing when you're working from _within_ the phylogenetic directory unless you are really aware of what  `nextstrain-pathogen.yaml` does.

**Comment by tsibley** [comment]
> Yeah, relative to the provided <directory> was my expectation, so I'd support this change.

Great!

> And I think this both _was_ the situation before the recent changes related to separate analysis directories, and still _is_ the case if we don't have a `nextstrain-pathogen.yaml`.

Yep, as the help still says.

Besides `--download`, this also affects `--exclude-from-download` and `--exclude-from-upload`. It should be doable to adjust all of them internally so that their interface is consistent regardless of `nextstrain-pathogen.yaml`.

> However we still need to convey what's going on here via help/docs messages

For sure. That's still the remit of this issue.
